addon/-private/system/model/internal-model.js

```diff
@@ -95,20 +95,18 @@ export default function InternalModel(type, id, store, _, data) {
   this.store = store;
   this._data = data || new EmptyObject();
   this.modelName = type.modelName;
-  this.dataHasInitialized = false;
+  this._loadingPromise = null;
```

Comment:
it is written to elsewhere;   i have added it here to preserve shape.

Really though we should add a setter and not have other parts of the system write private fields